supervisor/supervisor.go

```diff
@@ -141,24 +141,24 @@ func (sv *Supervisor) getHyperPod(container string, spec *specs.Spec) (hp *Hyper
 		return nil, fmt.Errorf("The container %s is already existing", container)
 	}
 	for _, ns := range spec.Linux.Namespaces {
-		if ns.Path != "" {
-			if strings.Contains(ns.Path, "/") {
-				return nil, fmt.Errorf("Runv doesn't support path to namespace file, it supports containers name as shared namespaces only")
+		if ns.Type == "network" && len(ns.Path) > 0 {
+			pidexp := regexp.MustCompile(`/proc/(\d+)/ns/net`)
+			matches := pidexp.FindStringSubmatch(ns.Path)
+			if len(matches) != 2 {
+				return nil, fmt.Errorf("Can't find shared container with network ns path %s", ns.Path)
 			}
-			if ns.Type == "mount" {
-				// TODO support it!
-				return nil, fmt.Errorf("Runv doesn't support shared mount namespace currently")
-			}
-			shared := ns.Path
-			cnt, ok := sv.Containers[shared]
-			if !ok {
-				return nil, fmt.Errorf("The container %s is not existing to share with", shared)
+			pid := matches[1]
```

Comment:
`pid` is only used for comparation, e.g. `pid == fmt.Sprintf("%d", c.ownerPod.getNsPid())`

So I think either converting `pid` to int or converting `c.ownerPod.getNsPid()` to string could work fine, they are basically the same :smile: